### PR TITLE
chore: mocknet config

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description
This pr adds mocknet configuration to the binary

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: x/observer/types/core_params_mainnet.go [hunk 1]
```diff
@@ -3,9 +3,81 @@
...
+	"github.com/zeta-chain/zetacore/common"
+)
+
 func GetCoreParams() CoreParamsList {
 	params := CoreParamsList{
-		CoreParams: []*CoreParams{},
+		CoreParams: []*CoreParams{
+			{
+				ChainId:                     common.EthChain().ChainId,
+				ConfirmationCount:           6,
```
Comment (CharlieMc0) [inline x/observer/types/core_params_mainnet.go:18]: This should be higher - 14
https://help.coinbase.com/en/coinbase/getting-started/crypto-education/glossary/confirmations

File: x/observer/types/core_params_mainnet.go [hunk 2]
```diff
@@ -3,9 +3,81 @@
...
+				InTxTicker:                  12,
+				OutTxTicker:                 15,
+				WatchUtxoTicker:             0,
+				GasPriceTicker:              30,
+				OutboundTxScheduleInterval:  30,
+				OutboundTxScheduleLookahead: 60,
+			},
+			{
+				ChainId:                     common.BscMainnetChain().ChainId,
+				ConfirmationCount:           6,
```
Comment (CharlieMc0) [inline x/observer/types/core_params_mainnet.go:31]: This should be higher - 14
https://help.coinbase.com/en/coinbase/getting-started/crypto-education/glossary/confirmations

File: x/observer/types/core_params_mainnet.go [hunk 3]
```diff
@@ -3,9 +3,81 @@
...
+				InTxTicker:                  5,
+				OutTxTicker:                 15,
+				WatchUtxoTicker:             0,
+				GasPriceTicker:              30,
+				OutboundTxScheduleInterval:  30,
+				OutboundTxScheduleLookahead: 60,
+			},
+			{
+				ChainId:                     common.ZetaChain().ChainId,
+				ConfirmationCount:           3,
```
Comment (CharlieMc0) [inline x/observer/types/core_params_mainnet.go:44]: This should be higher. Between 6-10. I've seen both numbers used for Cosmos

Comment (kingpinXD) [reply x/observer/types/core_params_mainnet.go:44]: This is not used

File: common/default_chains_mainnet.go [hunk 4]
```diff
@@ -20,7 +20,7 @@ func BscMainnetChain() Chain {
 func ZetaChain() Chain {
 	return Chain{
 		ChainName: ChainName_zeta_mainnet,
-		ChainId:   101,
+		ChainId:   7000,
```
Comment (brewmaster012) [inline common/default_chains_mainnet.go:23]: 70000

File: common/default_chains_mock_mainnet.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		ChainId:   137,
+	}
+}
+
+func DefaultChainsList() []*Chain {
+	chains := []Chain{
+		BtcMainnetChain(),
+		BscMainnetChain(),
+		EthChain(),
+		ZetaChain(),
```
Comment (brewmaster012) [inline common/default_chains_mock_mainnet.go:46]: i don't think this is used anywhere, having it here is confusing

File: x/observer/types/core_params_mainnet.go (hunk 3 above)
Comment (brewmaster012) [reply x/observer/types/core_params_mainnet.go:44]: again, this is not used, let's remove it

File: x/observer/types/core_params_mock_mainnet.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+				Erc20CustodyContractAddress: "",
+				InTxTicker:                  5,
+				OutTxTicker:                 15,
+				WatchUtxoTicker:             0,
+				GasPriceTicker:              30,
+				OutboundTxScheduleInterval:  30,
+				OutboundTxScheduleLookahead: 60,
+			},
+			{
+				ChainId:                     common.ZetaChain().ChainId,
```
Comment (brewmaster012) [inline x/observer/types/core_params_mock_mainnet.go:43]: let's remove it

File: zetaclient/config/config_mainnet.go [hunk 7]
```diff
@@ -42,13 +40,28 @@ func GetERC20CustodyABI() string {
...
+		BitcoinConfig:   BitcoinConfig,
+		ChainsEnabled:   []common.Chain{},
 	}
 }
+
+var BitcoinConfig = &BTCConfig{}
+
+var evmChainConfigs = map[int64]*EVMConfig{
+
+	common.ZetaChain().ChainId: {
```
Comment (brewmaster012) [inline zetaclient/config/config_mainnet.go:58]: this should be removed?

File: zetaclient/config/config_mock_mainnet.go [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+		BitcoinConfig:   BitcoinConfig,
+		ChainsEnabled:   []common.Chain{},
+	}
+}
+
+var BitcoinConfig = &BTCConfig{}
+
+var evmChainConfigs = map[int64]*EVMConfig{
+
+	common.ZetaChain().ChainId: {
```
Comment (brewmaster012) [inline zetaclient/config/config_mock_mainnet.go:58]: should remove this

**Review by brewmaster012 — CHANGES_REQUESTED** [review]

File: common/default_chains_mock_mainnet.go (hunk 5 above)
Comment (kingpinXD) [reply common/default_chains_mock_mainnet.go:46]: The value is used for this. 
https://github.com/zeta-chain/zeta-node/blob/8a842c0147927b5db627457effd363b1bb8ba475/zetaclient/evm_client.go#L695-L699

We can defeinitly remove it, but it should be done in a separate pr; this is for adding the config only.

File: common/default_chains_mainnet.go (hunk 4 above)
Comment (kingpinXD) [reply common/default_chains_mainnet.go:23]: This is for mainnet , added 70000 for mock mainnet

**Review by brewmaster012 — APPROVED** [review]

**Review by lumtis — COMMENTED** [review]
Just wondering, is there a different in the configuration compared to the actual mainnet one?
Don't we use mock mainnet to have an environment as close as mainnet as possible?

**Comment by kingpinXD** [comment]
> onfiguration compared to the actual mainnet one?
> Don't we use mock mainnet to have an environment as close as mainnet as possible?

Its the same , just the chain ID is different , 70000 vs 7000

**Review by lumtis — APPROVED** [review]

**Comment by lumtis** [comment]
@kingpinXD we can merge it now right?

**Pull request merged**